libs/nmea0183/src/sentence.cpp

```diff
@@ -722,3 +723,51 @@ const SENTENCE& SENTENCE::operator += ( LATLONG& source )
 
    return( *this );
 }
+
+/**
+ * @brief Convert a unicode wxString to a 7-bit NMEA compliant wxString.
+ * Any non compliant character is replaced by '_' character.
+ *
+ * @param field_string The string to be converted
+ * @return NMEA compliant string
+*/
+wxString SENTENCE::ToNmeaString(const wxString &field_string)
+{
+   // Use ToAscii() convert the input string to 7-bit ASCII. This method does
+   // not make clever conversion, it only replaces any no ASCII character by '_'.
+   wxString ascii_string = field_string.ToAscii();
+   int str_length = ascii_string.length();
+
+   // Loop throught the ASCII string to replace any NMEA reserved char by '_'
+   // Valid NMEA characters can be found in chapter 6.1 of NMEA0183 standard V3.01
+   char c;
+   for (int i = 0; i < str_length; i++)
+   {
+      c = ascii_string[i];
+      
+      if (c < 0x20) // No characters below 0x20 are valid
+      {
+         ascii_string[i] = '_';
+      }
+      else if (c > 0x7d) // No characters above 0x7d are valid
+      {
+         ascii_string[i] = '_';
+      }
+      else
+      {
+         switch ((unsigned char) ascii_string[i])
+         {
+            case 0x21: // '!'
```

Comment:
nit-picking:  Why not  like below, possibly after casting the char constant?
```
  case '$':
  case '*':
  ....
```

Reply:
No good reason to explain that.

Reply:
I could modify this but I just realized that the second NMEA related pull request I just submitted is starting from this one and it has a common file in the list of modified sources...